### PR TITLE
Task 32 & 96 | Flat Roof |  algorithm & testing

### PR DESCRIPTION
**Pull request by Runi-VN**
## Information
[Task#32 - Flat roof algorithm](https://tree.taiga.io/project/maltemagnussen-fog/task/96?kanban-status=1592822)

[Task#96 - Testing of #32 algorithm](https://tree.taiga.io/project/maltemagnussen-fog/task/96?kanban-status=1592822)  

This algorithm is basically just an adapted version of the [supplied PDF partslist](https://datsoftlyngby.github.io/dat2sem2019Spring/Modul4/Fog/CP01_DUR.pdf).  

**Important**: I updated maven version for lambda expressions: https://github.com/HrBjarup/Fog-Carport/pull/75/commits/3693ef0f034e5c043ee9afb1f382ce0aa6f702a2

**Important**: This pull request contains changes to the DML script

### Process
The algorithm changed a lot underway. [I saved some of the legacy algorithm](https://gist.github.com/Runi-VN/593bb40aa1224573706647f838b7aacc) for future use. @Castau and I want to develop a help method that can reduce code count for both of our algorithms.
#### Resources
Plastic tile calculation: Plastic tile calculation 

### Things that are not as they should be:
* ScrewCounter is not working at the moment. (Will implement at a later point)
* PartsListModel does not have relevant fields for SVG-drawings.
* The algorithm is (with documentation) nearly 600 lines long. This needs to be shortened, and it can -- a lot of the calculations are _basically_ the same.
* Currently throws 'loginException' (**Suggestion**: We _all_ throw loginException; rename it instead of creating a new -- then create a new, actual _loginException_ (saves time))
* Tagpap has not been made yet. (Not part of user story)
  * If we decide to add this, then all functionality is ready; just need a simple calculation.
* The tests are not covering all edge cases. Priority?
* Helptext is currently manually set. This should be removed for upcoming database addition.
### Other:
* Rafter only calculates for one length. (6000mm) 
  * This is how Fog does it too. 
    * Added material (4800mm, 3600mm) to fix it, but didnt apply due to it being outside task scope.

### CODE REVIEW

File: FogCarport/src/main/java/data/models/PartslistModel.java [hunk 1]
```diff
@@ -112,34 +118,39 @@ public void setPostPosRear(ArrayList postPosRear)
...
         hash = 79 * hash + Objects.hashCode(this.billOfMaterials);
         hash = 79 * hash + this.totalprice;
         return hash;
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
+    public boolean equals(Object obj)
+    {
```
Comment (HrBjarup) [inline FogCarport/src/main/java/data/models/PartslistModel.java]: I like those changes ;P

Comment (Runi-VN) [reply FogCarport/src/main/java/data/models/PartslistModel.java]: lmao. I had 42 files changed when merging developer into this branch -- due to us using different formatting. @_@

Comment (MalteMagnussen) [reply FogCarport/src/main/java/data/models/PartslistModel.java]: Don't tell Camilla

File: FogCarport/src/main/java/logic/Calculations/RoofFlatCalc.java [hunk 2]
```diff
@@ -26,51 +25,68 @@
...
+    private int fittingRightID = 15;
+    private int rafterSmallID = 5; // 45x195x4800 (UNUSED CUSTOM ITEM (OUTSIDE SCOPE))
+    private int rafterLargeID = 54; //45x195x6000
+    private int fasciaWidthBottomID = 55; //25x200x3600
+    private int fasciaLengthBottomID = 56; //25x200x5400
+    private int fasciaWidthTopID = 57; //25x125x3600
+    private int fasciaLengthTopID = 58; //25x125x5400
+    private int bargeboardLengthID = 59;
+    private int bargeboardWidthID = 60;
+    private int bargeboardScrewsID = 20; //200 a pack
```
Comment (MalteMagnussen) [inline FogCarport/src/main/java/logic/Calculations/RoofFlatCalc.java]: Nice. I also did this for the Shed. Today at the meeting he said we should somehow put all this in a seperate class? Didn't quite get that part. But nice that you've taken the time to set it up like this aswell with the fields.

Comment (Runi-VN) [reply FogCarport/src/main/java/logic/Calculations/RoofFlatCalc.java]: Yes, AKA said a possible solution would be a seperate class with constants for the IDs.

It is still a solution, but I think it is out of scope. This is something we can mention in our report.
